jans-cedarling/cedarling/src/authz/mod.rs

```diff
@@ -132,45 +122,54 @@ impl Authz {
 
         // Convert [`AuthorizeEntitiesData`] to  [`cedar_policy::Entities`] structure,
         // hold all entities that will be used on authorize check.
-        let entities = entities_data.entities(Some(&schema.schema))?;
+        let entities = entities_data.clone().entities(Some(&schema.schema))?;
 
         // Check authorize where principal is `"Jans::Workload"` from cedar-policy schema.
-        let workload_result: Option<Response> = if self.config.authorization.use_workload_principal
-        {
-            match self.execute_authorize(ExecuteAuthorizeParameters {
-                entities: &entities,
-                principal: principal_workload_uid.clone(),
-                action: action.clone(),
-                resource: resource_uid.clone(),
-                context: context.clone(),
-            }) {
-                Ok(resp) => Some(resp),
-                Err(err) => return Err(AuthorizeError::CreateRequestWorkloadEntity(err)),
-            }
+        let (workload_authz_result, workload_authz_info, workload_entity_claims) = if self.config.authorization.use_workload_principal {
+            let principal = entities_data.workload.uid();
+
+            let authz_result = self.execute_authorize(ExecuteAuthorizeParameters { entities: &entities, principal: principal.clone(), action: action.clone(), resource: resource_uid.clone(), context: context.clone() }).map_err(AuthorizeError::WorkloadRequestValidation)?;
+
+            let authz_info = WorkloadAuthorizeInfo {
+                principal: principal.to_string(),
+                diagnostics: Diagnostics::new(
+                    authz_result.diagnostics(),
+                    &self.config.policy_store.policies,
+                ),
+                decision: authz_result.decision().into(),
+            };
+
+            let workload_entity_claims = get_entity_claims( self.config.authorization.decision_log_workload_claims.as_slice(),&entities, principal);
+            (Some(authz_result), Some(authz_info), Some(workload_entity_claims))
         } else {
-            None
+            (None, None, None)
         };
 
         // Check authorize where principal is `"Jans::User"` from cedar-policy schema.
-        let person_result: Option<Response> = if self.config.authorization.use_user_principal {
-            match self.execute_authorize(ExecuteAuthorizeParameters {
-                entities: &entities,
-                principal: principal_user_entity_uid.clone(),
-                action: action.clone(),
-                resource: resource_uid.clone(),
-                context: context.clone(),
-            }) {
-                Ok(resp) => Some(resp),
-                Err(err) => return Err(AuthorizeError::CreateRequestUserEntity(err)),
-            }
+        let (user_authz_result, user_authz_info, user_entity_claims) = if self.config.authorization.use_user_principal {
```

Comment:
It looks really weird, return all this structs here but here we only check authorize result for user...
Same with previous..